### ISSUE TITLE
feat: EPubLoader

### ISSUE THREAD

**Issue by 0xMochan** [0xPlaygrounds/rig#160]
- [x] I have looked for existing issues (including closed) about this

## Feature Request
Add a loader type that can work with the EPUB format (ebook format).

### Motivation
Oftentimes, PDFs are unreliable format for loading. While it is the most common and likely will be continued to be used, adding more loader types adds flexibilities in people's abilities to construct RAGs.

### Proposal
- [ ] Create a new loader type (`rig-core/src/loaders/epub.rs`) `EpubFileLoader`

### Alternatives
- Use a Epub -> PDF converter before using `PdfFileLoader`

**Comment by danik-tro** [comment]
@0xMochan Hi, I would like to contribute and start working on this feature. Please let me know if I can proceed.

**Comment by 0xMochan** [comment]
> [@0xMochan](https://github.com/0xMochan) Hi, I would like to contribute and start working on this feature. Please let me know if I can proceed.

Yes, this feature is a great way to get started on contributing to the repo! Let me know if you run into any issues.

**Comment by danik-tro** [comment]
> Yes, this feature is a great way to get started on contributing to the repo! Let me know if you run into any issues.

Great, I’ll start working on it. Thank you!

**Comment by joshua-mo-143** [comment]
Closing because the relevant PR to close this has been merged.
